ann_benchmarks/__init__.py

```diff
@@ -213,18 +213,28 @@ def __init__(self, metric, P):
         self._metric = metric
 
     def fit(self, X):
+        os.environ['OMP_THREAD_LIMIT'] = '40'
```

Comment:
This is only for index construction which follows existing code of NmslibReuseIndex.  Search is still single threaded  (I'll post about construction time separately.)  
Regarding search time parallelization, KGraph batch search achieves 3.4x speedup on a 4 core CPU with hyperthreading on the glove dataset.